src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -18,17 +18,34 @@ public class PackageSourceProvider : IPackageSourceProvider
 
         public PackageSourceProvider(
           ISettings settings)
-            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources)
+            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources, enablePackageSourcesChangedEvent: true)
+        {
+        }
+
+        public PackageSourceProvider(
```

Comment:
Does anyone have any ideas how we can discourage customers from using this overload, given NuGet/Home#8479 should remove the need for it?

My best idea is to mark it as obsolete, but then we'll get warnings where we use it unless we disable the obsolete warning for those lines, which doesn't feel better than not doing anything.

Reply:
Warn in XML doc comments?

Reply:
mark as obsolete, say it's going away, suppress on our end, fix in 6.0? 
That's the best I have. 

Also, you can't not disable the warning, we have WarningsAsErrors. :)